java/src/main/java/com/microsoft/onnxruntime/ONNXUtil.java

```diff
@@ -0,0 +1,648 @@
+     * @return A float array.
+     */
+    public static Object newFloatArray(long[] shape) {
+        switch (shape.length) {
+            case 1:
+                return new float[(int) shape[0]];
+            case 2:
+                return new float[(int) shape[0]][(int) shape[1]];
+            case 3:
+                return new float[(int) shape[0]][(int) shape[1]][(int) shape[2]];
+            case 4:
+                return new float[(int) shape[0]][(int) shape[1]][(int) shape[2]][(int) shape[3]];
+            case 5:
+                return new float[(int) shape[0]][(int) shape[1]][(int) shape[2]][(int) shape[3]][(int) shape[4]];
+            case 6:
+                return new float[(int) shape[0]][(int) shape[1]][(int) shape[2]][(int) shape[3]][(int) shape[4]][(int) shape[5]];
+            case 7:
+                return new float[(int) shape[0]][(int) shape[1]][(int) shape[2]][(int) shape[3]][(int) shape[4]][(int) shape[5]][(int) shape[6]];
+            case 8:
+                return new float[(int) shape[0]][(int) shape[1]][(int) shape[2]][(int) shape[3]][(int) shape[4]][(int) shape[5]][(int) shape[6]][(int) shape[7]];
+            default:
+                throw new IllegalArgumentException("Arrays with less than 1 and more than 8 dimensions are not supported.");
+        }
+    }
+
+    /**
+     * Creates a new primitive double array of up to 8 dimensions, using the supplied shape.
+     * <p>
+     * Does not check the shape to see if all it's elements are positive.
+     *
+     * @param shape The shape of array to create.
+     * @return A double array.
+     */
+    public static Object newDoubleArray(long[] shape) {
+        switch (shape.length) {
+            case 1:
+                return new double[(int) shape[0]];
+            case 2:
+                return new double[(int) shape[0]][(int) shape[1]];
+            case 3:
+                return new double[(int) shape[0]][(int) shape[1]][(int) shape[2]];
+            case 4:
+                return new double[(int) shape[0]][(int) shape[1]][(int) shape[2]][(int) shape[3]];
+            case 5:
+                return new double[(int) shape[0]][(int) shape[1]][(int) shape[2]][(int) shape[3]][(int) shape[4]];
+            case 6:
+                return new double[(int) shape[0]][(int) shape[1]][(int) shape[2]][(int) shape[3]][(int) shape[4]][(int) shape[5]];
+            case 7:
+                return new double[(int) shape[0]][(int) shape[1]][(int) shape[2]][(int) shape[3]][(int) shape[4]][(int) shape[5]][(int) shape[6]];
+            case 8:
+                return new double[(int) shape[0]][(int) shape[1]][(int) shape[2]][(int) shape[3]][(int) shape[4]][(int) shape[5]][(int) shape[6]][(int) shape[7]];
+            default:
+                throw new IllegalArgumentException("Arrays with less than 1 and more than 8 dimensions are not supported.");
+        }
+    }
+
+    /**
+     * Reshapes an int array into the desired n-dimensional array.
```

Comment:
>Reshapes an in [](start = 7, length = 14)

So does this follow some specification? For example, does it follow standard ONNX Reshape?

Reply:
It's the n-dimensiomal version of row major order. It's used in the unit tests so I think it's what the runtime uses.

Reply:
Please, add row major order into comments.

---
In reply to: [340896273](https://github.com/microsoft/onnxruntime/pull/2215#discussion_r340896273) [](ancestors = 340896273)

Reply:
Will do.

Reply:
Done.